src/interactions/slash_commands/Key.ts

```
@@ -0,0 +1,167 @@
+        const member = await memberService.findMemberByUserId(user.id);
+
+        const command = interaction.options.getSubcommand();
+        switch (command) {
+            case "add": {
+                if (!member) {
+                    throw new Error(
+                        `You are not a registered RTI member. Please apply to RTI first.`
+                    );
+                }
+
+                if (member.gw2ApiKey) {
+                    const prompt = `You already have an API Key saved. Do you want to overwrite it?`;
+                    const overwrite = new YesNoDialog(prompt, user, channel);
+                    await interaction.reply({
+                        content: "Answer the question below.",
+                        ephemeral: true,
+                    });
+                    const cont = await overwrite.run();
+                    overwrite.deleteMessages(5000);
+
+                    if (!cont) {
+                        await interaction.followUp({
+                            content: "Cancelled.",
+                            ephemeral: true,
+                        });
+                        return;
+                    }
+                }
+
+                const token = interaction.options.getString("key", true);
+                const verification =
+                    await App.instance().gw2ApiService.verifyApiToken(
+                        token,
+                        member.gw2Name
+                    );
+                const result = await memberService.updateMemberApiKey(
+                    member.userId,
+                    token
+                );
+
+                if (!verification.valid) {
+                    await interaction.reply({
+                        content: `${verification.msg ?? `Unknown error.`}`,
+                        ephemeral: true,
+                    });
+                    return;
+                }
+
+                if (!result.success) {
```

You should throw an error here.